scripts/check_license_headers.sh

```diff
@@ -0,0 +1,11 @@
+set -e
+echo "Checking that a Apache 2 license header is present at the top of all scala files in src/"
+
+src_files_with_missing_license=$(find src -type f -name '*.scala' -exec bash -c '[ ! -z "$(head -1 {} | grep -v "// Licensed to the Apache Software Foundation (ASF) under one")" ] && echo "{}"' \;)
```

Comment:
This could be simplified to:

```bash
src_files_with_missing_license="$(find src -type f -name "*.scala" -exec grep -LF "Licensed to the Apache Software Foundation (ASF)" {} \+)"
```

The `-F` option to `grep` isn't strictly necessary, but it will make it use a faster algorithm for searching on fixed strings.

**Note** The semantics of this are _slightly_ different to your code, which checks for the string on the first line of the file; mine checks for its (lack of) existence anywhere in the file.

Reply:
Thanks for sharing about `-F` :)